### ISSUE TITLE
[Bug] Fragments on fields and subfields not being merged correctly

### ISSUE THREAD

**Issue by smkhalsa** [neo4j-graphql/neo4j-graphql-js#180]
I'm having trouble using fragments on both fields and subfields in the same query. For example, when I make this query...

```
fragment ArtistFragment on Artist {
  albums {
    name
    songs {
      name
    }
  }
}
fragment AlbumFragment on Album {
  id
  name
  imageName
}
query {
  Artist(first:1) {
    ...ArtistFragment
    albums {
      ...AlbumFragment
    }
  }
}
```

... I get the following error:
```
"Resolve function for \"Album.songs\" returned undefined"
```
NOTE: If I don't use query fragments, the query runs fine. 

Here's what my schema looks like:

```
type Album implements {
  id: ID!
  name: String!
  imageName: String
  artists: [Artist!]! @relation(name: "BY_ARTIST", direction: "OUT")
  songs: [Song!]! @relation(name: "HAS_SONG", direction: "OUT")
}

type Artist {
  id: ID!
  name: String!
  imageName: String
  albums: [Album!]! @relation(name: "BY_ARTIST", direction: "IN")
  songs: [Song!]! @relation(name: "BY_ARTIST", direction: "IN")
}

type Song {
  id: ID!
  name: String!
  imageName: String
  fileName: String!
  artists: [Artist!]! @relation(name: "BY_ARTIST", direction: "OUT")
  albums: [Album!]! @relation(name: "HAS_SONG", direction: "IN")
}
```

**Comment by smkhalsa** [comment]
@cooperka any ideas on this from your work on #78?

**Comment by cooperka** [comment]
At a quick glance, this is supposed to resolve to:

```
Artist(first:1) {
    albums {
        name
        songs {
            name
        }
    }
    albums {
        id
        name
        imageName
    }
}
```

does that exact query (with duplicate `albums`) run as expected? If so, there's probably a bug somewhere in the parsing implementation.

**Comment by smkhalsa** [comment]
@cooperka actually that query gives me the same error.

However, if I roll the queries together it runs fine:

```
Artist(first:1) {
    albums {
        id
        name
        imageName
        songs {
          name
        }
    }
}
```

**Comment by cooperka** [comment]
The problem is likely the duplication, rather than the fragments. It still may be an implementation bug, but hopefully that narrows it down.

As a workaround, you might try modifying your fragments to prevent the duplication. I hope that helps, unfortunately I don't have the time now to dig deeper.

**Comment by imkleats** [comment]
I would agree with @cooperka that this is likely due to duplication of field names within the query. The solution proposed in the GraphQL [documentation](https://graphql.github.io/learn/queries/) is to use an alias like this:

`query {
  Artist(first:1) {
    ...ArtistFragment
    albumAlias: albums {
      ...AlbumFragment
    }
  }
}`

Let us know if that works for you. I'm not aware that query de-duplication is a feature that this package is intended to support yet, which might make this less of a bug than a new feature.

**Comment by smkhalsa** [comment]
@imkleats I agree that this seems to be related to using the same field in multiple parts of a query / fragment. However, it appears to be a bug in the neo4j-graphql-js implementation, not a limitation of the graphql spec. The section on aliases in the graphql [documentation](https://graphql.github.io/learn/queries/#aliases) you are referencing says `you can't directly query for the same field with different arguments`. That isn't the issue here - I'm not using any arguments on the fields in question, and in fact if I try using an alias as you suggest, I still get the same error.

**Comment by smkhalsa** [comment]
This issue keeps coming up for me, especially when using nested fragments. There seems to be an issue with how fields from fragments are getting merged with other fields in the query. 

@johnymontana @michaeldgraham any ideas?

**Comment by michaeldgraham** [comment]
https://github.com/neo4j-graphql/neo4j-graphql-js/issues/608
